src/main/java/core/basesyntax/strategy/handler/DecrementHandler.java

```diff
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.handler;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.service.OperationHandler;
+
+public abstract class DecrementHandler implements OperationHandler {
+    @Override
+    public void apply(Fruit fruit, int quantity) {
+        fruit = Storage.fruits.get(fruit.getName());
+        if (fruit == null || fruit.getQuantity() < quantity) {
+            throw new RuntimeException("Not enough fruit available.");
+        }
+        fruit.subtractQuantity(quantity);
```

Comment:
There is no 'subtractQuantity' method in the 'Fruit' class according to the provided files. Ensure that the 'Fruit' class has the appropriate method to decrease the quantity or consider updating the storage with the new quantity.